source/test_env.cpp

```diff
@@ -58,7 +64,13 @@ void notify_performance_coefficient(const char* measurement_name, const double v
 
 void notify_completion(bool success)
 {
-    printf("{{%s}}" NL "{{%s}}" NL, success ? TEST_ENV_SUCCESS : TEST_ENV_FAILURE, TEST_ENV_END);
+    printf("{{%s}}" NL, success ? TEST_ENV_SUCCESS : TEST_ENV_FAILURE);
+#ifdef YOTTA_CFG_DEBUG_OPTIONS_COVERAGE
+    coverage_report = true;
+    gcov_exit();
+    coverage_report = false;
```

Comment:
I'm afraid that if you disable interrupts you might prevent the serial port from working properly (in the future :) ). But yes, I see your point, this looks safe enough for now. I just hope it doesn't come back to bite us at some point.